lib/pf/util.pm

```diff
@@ -1352,6 +1353,43 @@ sub ping {
     return $p->ping($host);
 }
 
+=head2 running_as
+
+Returns the user which the current process is running as
+
+=cut
+
+sub running_as {
+    return $ENV{LOGNAME} || $ENV{USER} || getpwuid($<);
+}
+
+=head2 run_as_pf
+
+Sets the UID and GID of the currently running process to pf
+
+=cut
+
+sub run_as_pf {
+    # Early return if we're already running as pf
+    return $TRUE if(running_as eq "pf");
+
+    my (undef, undef,$uid,$gid) = getpwnam('pf');
```

Comment:
Just compare $< to $uid instead of using running_as since the environmental variables could be modified.